networkx/algorithms/centrality/domirank.py

```diff
@@ -0,0 +1,342 @@
+
+    DomiRank centrality [1]_ computes the centrality for a node by aggregating
+    1 minus the centrality of each node in its neighborhood. This essentially finds the
+    dominance of a node in its neighborhood, where the parameter $\alpha$ determines
+    the amount of competition in the system by modulating $\sigma = \alpha/|\lambda_N|$.
+    The competition parameter $\alpha$ tunes the balance of DomiRank centrality's
+    integration of local and global topological information, to find nodes that are either
+    locally or globally dominant. It is important to note that for the iterative formulation
+    of DomiRank (as seen below) the competition parameter is bounded: $\sigma \in (0,1/|\lambda_N|]$.
+    The DomiRank centrality of node $i$ is defined as the stationary solution to the dynamical system:
+
+    .. math::
+
+        \,d\Gamma_i(t)/dt = \sigma (d_i - \sum_j A_{ij} \Gamma_j(t)) - \Gamma_i(t),
+
+    where $A$ is the adjacency matrix, $\lambda_N$ its smallest eigenvalue, and $d_i$ is the degree of node $i$.
+    Note that the definition presented here is valid for unweighted, weighted,
+    directed, and undirected networks, so in the more general case,
+    a non-zero entry of the adjacency matrix $A_{ij}=w_{ij}$
+    represents the existence of a link from node $i$ to node $j$
+    with a weight $w_{ij}$. The steady state solution to this equation
+    is computed using Newton's method. In general, one will notice that important
+    nodes identified by DomiRank will be connected to a lot of other,
+    unimportant nodes. However, positionally important nodes
+    can also be dominated by joint-dominance of nodes, that work together
+    in order to dominate the positionally important node. This centrality
+    gives rise to many interesting emergent phenomena;
+    see [1]_ for more information.
+    DomiRank centrality can also be
+    expressed in its analytical form, where the competition can now be
+    supercharged - i.e. $\sigma \in [0,+\infty)$. The analytical equation
+    takes the form:
+
+    .. math::
+
+        \boldsymbol{\Gamma} = \sigma (\sigma A + I_{N\times N})^{-1}  A  \boldsymbol{1}_{N\times 1},
+
+    where $I$ is the identity matrix. This analytical equation can be solved
+    as a linear system.
+
+    DomiRank tends to have only positive values for relatively low
+    competition levels ($\sigma \to 0$). However as the competition
+    level increases, negative values might emerge. Nodes with negative
+    dominance represent completely submissive nodes, which instead
+    of fighting for their resources/dominance, directly give up these
+    resources to their neighborhood.
+
+    Finally, DomiRank centrality does not require the network to be weakly or strongly
+    connected, and can be applied to networks with many components.
+
+    Parameters
+    ----------
+    G : graph
+        A NetworkX graph.
+
+    analytical: string, optional (default="iterative")
```

Comment:
```suggestion
    method: string, optional (default="iterative")
```

You didn't change the name of the parameter here. I prefer `method`; `comp_method` makes me feel like you're comparing something, when that isn't the case.